src/components/navigation/Navigation.tsx

```diff
@@ -28,6 +41,21 @@ export default function Navigation() {
             <Link href={`/${link.path}`}>{link.name}</Link>
           </li>
         ))}
+        {!loading &&
```

Comment:
NIT: it's probably worth to add a proper loader here to account for "loading" in a better way. Also, by design, most of the UIs usually cache the user so that it is immediately available after the signup/login.